### PR TITLE
Update setup commands and docs

### PR DESCRIPTION
**Pull request by nmartin413**
- updates some of the Makefile commands
- adds a (preferred) option to run in a docker shell

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,11 +1,18 @@
+cur_dir   := $(shell basename $(CURDIR))
+mount_dir := $(PWD)
+
 # Targets for local development
+init::    git_reset
+shell::   docker_shell
```
Comment (darwayne) [inline Makefile:6]: woah .. this is fancy!

**Comment by darwayne** [comment]
going to pull down to validate changes FYI

**Review by darwayne — COMMENTED** [review]
I'm seeing the following when I run make install

```
$ make install
python3.9 -m venv venv
. venv/bin/activate
python -m pip install --upgrade pip
Requirement already satisfied: pip in ~/.pyenv/versions/3.9.11/lib/python3.9/site-packages (22.0.4)
Collecting pip
  Using cached pip-22.2.2-py3-none-any.whl (2.0 MB)
Installing collected packages: pip
  Attempting uninstall: pip
    Found existing installation: pip 22.0.4
    Uninstalling pip-22.0.4:
      Successfully uninstalled pip-22.0.4
Successfully installed pip-22.2.2
pip3 install -qr requirements.txt
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
panther-analysis-tool 0.15.1 requires aiohttp, which is not installed.
```
- image


I thought the docker_shell stuff would have commands run through docker?

File: Makefile [hunk 2]
```diff
@@ -17,9 +24,20 @@ ci_lint::
...
 utl_activate: venv
 	. venv/bin/activate
+
+venv_rm:
+	rm -rf venv
+
+docker_shell:
+	docker run --rm -it -v "$(mount_dir):/$(cur_dir)" --workdir "/$(cur_dir)" python:3.9 /bin/bash
+
+git_reset:
```
Comment (darwayne) [inline Makefile:40]: we should probably add a confirmation prompt here

**Review by danbiwer — APPROVED** [review]

File: Makefile [hunk 3]
```diff
@@ -17,9 +24,20 @@ ci_lint::
...
+	python3.9 -m venv venv
 
 utl_activate: venv
 	. venv/bin/activate
+
+venv_rm:
+	rm -rf venv
+
+docker_shell:
+	docker run --rm -it -v "$(mount_dir):/$(cur_dir)" --workdir "/$(cur_dir)" python:3.9 /bin/bash
```
Comment (maxrichie5) [inline Makefile:38]: Is there a way to automatically run the pip install command once inside the image to avoid making the user do it?

**Review by maxrichie5 — DISMISSED** [review]

**Review by darwayne — APPROVED** [review]
thought the make shell thingy made all commands use docker .. but after chatting with Nic, I now see that the intent is for the user to first run `make docker_shell` then run necessary make commands

File: Makefile (hunk 3 above)
Comment (nmartin413) [reply Makefile:40]: Yes, but i want to avoid having a custom docker image for this

**Pull request merged**